aedes.js

```diff
@@ -131,8 +127,18 @@ export class Aedes extends EventEmitter {
       )
     }, opts.heartbeatInterval * 4)
 
-    function receiveWills (chunks, done) {
-      that._parallel(that, checkAndPublish, chunks, done)
+    async function receiveWills (chunks, done) {
+      try {
+        await Promise.all(chunks.map(chunk => new Promise((resolve, reject) => {
+          checkAndPublish(chunk, (err) => {
+            if (err) reject(err)
+            else resolve()
+          })
+        })))
+        done()
+      } catch (err) {
+        done(err)
+      }
```

Comment:
This is by far not equivalent. It also creates a bazillion of promises and it will significantly impact performance